Add missing fields to installation

jastkand opened the pull request:
As per [Uploading installation data](https://parse.com/docs/rest/guide/#push-notifications-uploading-installation-data) section of Parse documentation installation object includes such fields as `deviceType`, `pushType`, `GCMSenderId`, `installationId` which were not listed in this gem.

As per documentation these fields are readonly, but they must be used to create new records, so I think they should be specified in `UPDATABLE_FIELDS` array.

Also I've added a method `Parse::Installation.build` which can be used to build new installation object using REST API. This method is just a sugar, because now you also could create installation objects using something like this:

```
Parse::Installation.new(nil, client).tap do |obj|
  obj['deviceToken'] = 'device_token_here'
  obj['deviceType'] = 'ios'
end
```

With this method it comes a little bit more nice

```
Parse::Installation.build(client).tap do |obj|
  obj['deviceToken'] = 'device_token_here'
  obj['deviceType'] = 'ios'
end
```

Also I've made some code improvements, removed explicit variable assignments and assignment in condition.

rhymes commented:
@jastkand thanks for the patch! I have two reservations though (which should be addressed by @Xavdidtheshadow  and @adelevie actually :-) ):

**a** - do we need another factory method? If so, maybe it's better to keep adding to the client object:

```
client.installation.tap do |installation|
 ...
end
```

like https://github.com/jastkand/parse-ruby-client/blob/add_device_type_to_installation/lib/parse/client.rb#L106 instead of creating another factory method in the originating class. Just a thought.

By the way we should also start harmonizing the names of different factory methods in different classes. We have:

```
Parse.init
Parse.create
Pointer.make
````

**b** - Why the Installation initializer asks for a objectId if it's not required anywhere else in the class? It seems to be required only for

```
def method
   @parse_object_id ? :put : :post
end
```

My real question is: are we going to break compatibility with the public interface and cleanup the methods? Because Installation.build seems to be another facade to just ameliorate the actual interface, so let's fix that instead...

jastkand commented:
@rhymes Thanks for the review. 

I think that having factory method for installation would be rather handy. If we have such method we will be able to do
```
client.installation.tap do |installation|
 ...
end
```
for creating installation, and
```
client.installation(object_id).get
```
for getting one. This seems to be nice.

`Parse::Installation.build` method looks handy but in fact it just tries to hide internal implementation of `Parse::Installation`. So it would be better to fix the interface of class.

jastkand commented:
I have updated pull-request with new factory method `installation` which was mentioned before. Also I've updated documentation with use of this method. `Parse::Installation.build` was removed as it is not really useful when we have factory method. Finally I've noticed that we are going to use Rubocop to check code-style of the gem, so I changed double quotes with single ones where is did changes.

@rhymes, @Xavdidtheshadow, what do you think?

rhymes commented:
I like it @jastkand thanks!

Xavdidtheshadow commented:
@rhymes @jastkand awesome, looks good! I'll merge it in.
